discojs/src/memory/base.ts

```diff
@@ -0,0 +1,56 @@
+import * as tf from '@tensorflow/tfjs'
+
+import { TaskID } from '..'
+import { ModelType } from './model_type'
+
+export abstract class Memory {
+  /**
+  * Fetches metadata of the model.
+  * @param taskID the working model's corresponding task
+  * @param modelName the working model's file name
+  */
+  abstract getModelMetadata (type: ModelType, taskID: TaskID, modelName: string): Promise<tf.io.ModelArtifactsInfo>
+
+  /**
+  * Loads the current working model and returns it as a fresh TFJS object.
+  * @param taskID the working model's corresponding task
+  * @param modelName the working model's file name
+  */
+  abstract getModel (type: ModelType, taskID: TaskID, modelName: string): Promise<tf.LayersModel>
+
+  /**
+ * Loads a model from the model library into the current working model.
+ * @param taskID the saved model's corresponding task
+ * @param modelName the saved model's file name
+ */
+  abstract loadSavedModel (taskID: TaskID, modelName: string): Promise<void>
+
+  /**
+ * Loads a fresh TFJS model object into the current working model.
+ * @param taskID the working model's corresponding task
+ * @param modelName the working model's file name
+ * @param model the fresh model
+ */
+  abstract updateWorkingModel (taskID: TaskID, modelName: string, model: tf.LayersModel): Promise<void>
+
+  /**
+ * Adds the current working model to the model library.
+ * @param taskID the working model's corresponding task
+ * @param modelName the working model's file name
+ */
+  abstract saveWorkingModel (taskID: TaskID, modelName: string): Promise<void>
+
+  /**
+ * Removes the model model.
```

Comment:
minor typo, 2x.
also you didn't document the `ModelType` param in this one and some above one. but fine anyway :)

Reply:
Typo fixed & `ModelType` commented :)